### ISSUE TITLE
Firestore plugin crashes on Android (NullPointerException)

### ISSUE THREAD

**Issue by rock3r** [flutter/flutter#14207]
## Steps to Reproduce

Build the [DartConf 1.1.0 Android app](https://github.com/rock3r/conference_app/releases/tag/dartconf18_android_v1.2.0), run on a Nexus 5X with Android 8.1.

I am not sure what the issue is nor how to reproduce it but this showed up on my Play console:

```
java.lang.NullPointerException: 
  at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$EventObserver.onEvent (CloudFirestorePlugin.java:137)
  at io.flutter.plugins.firebase.cloudfirestore.CloudFirestorePlugin$EventObserver.onEvent (CloudFirestorePlugin.java:119)
  at com.google.firebase.firestore.zzi.onEvent (Unknown Source:31)
  at com.google.android.gms.internal.zzejz.zza (Unknown Source:6)
  at com.google.android.gms.internal.zzeka.run (Unknown Source:6)
  at android.os.Handler.handleCallback (Handler.java:790)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6494)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:438)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:807)
```

## Logs

Proper logs N/A, this comes from production. `flutter analyze` and `flutter doctor` are fine.

**Comment by filiph** [comment]
Thanks for the bug! I briefly looked at the conference app code and it seems it's on `cloud_firestore` version 0.1.2, while the current `cloud_firestore` is now 0.2.6.

I suggest we upgrade the app to the latest version before looking deeper into this issue.

cc @devoncarew: Any particular reason why we shouldn't upgrade?

In case upgrading is not an option, here's the `cloud_firebase` java line that throws the NPE: https://github.com/flutter/plugins/blob/d63c7ab631beee7bbc4d2ee98284ccd7dcca684b/packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java#L137

**Comment by filiph** [comment]
(I just looked at the [0.2.6 version](https://github.com/flutter/plugins/blob/master/packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java#L156) of the Java method that throws and it's identical to 0.1.2.)

**Comment by rock3r** [comment]
I just noticed this still happens on 0.2.6; the new crash has the same stacktrace, but it's on a Pixel XL (Android 8.1)

**Comment by collinjackson** [comment]
It looks to me like this was fixed in flutter/plugins#376

**Comment by kroikie** [comment]
@rock3r

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/910. Any further collaboration will be done there.
